scripts/prune_lm_dir.py

```diff
@@ -328,6 +349,106 @@ def FinalizeOutput(final_work_out):
         os.remove(args.lm_dir_out + "/num_splits")
 
 
+def MatchTargetSize(num_ngrams):
+    return abs(num_ngrams - args.target_size) / float(args.target_size) < args.tolerance
+
+def AttempOnce(scale, step, attempt, recovery_step, recovery_size):
+    global logprob_changes, steps, final_num_ngrams
+
+    if step > 0:
+        steps += 'prune*{0} EM EM EM'.format(scale).split()
+
+    # Prune step
+    logprob_changes.append(RunStep(step, in_step=recovery_step))
+    step += 1
+    while step < len(steps): # EM steps
+      logprob_changes.append(RunStep(step))
+      step += 1
+    attempt += 1
+
+    if MatchTargetSize(final_num_ngrams):
+        return (True, step, attempt, recovery_step, recovery_size);
+
+    if attempt > args.max_attempt:
+        sys.exit("prune_lm_dir.py: Too many attempts, please set a higher threshold and rerun.")
+
+    # always prune from LM larger than target_size
+    if final_num_ngrams > args.target_size and (recovery_size <= 0 or final_num_ngrams < recovery_size):
+        recovery_step = step
+        recovery_size = final_num_ngrams
+
+    return (False, step, attempt, recovery_step, recovery_size)
+
+class LinearEstimator(object):
```

Comment:
Need some documentation of this.  Please provide examples of use, in a comment.
Think about potential for division by zero.